webapi/CopilotChat/Controllers/ChatHistoryController.cs

```diff
@@ -80,13 +80,14 @@ public async Task<IActionResult> CreateChatSessionAsync([FromBody] CreateChatPar
         var newChat = new ChatSession(chatParameter.Title);
         await this._sessionRepository.CreateAsync(newChat);
 
-        var initialBotMessage = this._promptOptions.InitialBotMessage;
-        // The initial bot message doesn't need a prompt.
+        // Create initial bot message
         var chatMessage = ChatMessage.CreateBotResponseMessage(
             newChat.Id,
-            initialBotMessage,
-            string.Empty);
+            this._promptOptions.InitialBotMessage,
+            string.Empty, // The initial bot message doesn't need a prompt.
+            new TokenUsage(0, 0));
         await this._messageRepository.CreateAsync(chatMessage);
+        newChat.InitialBotMessage = chatMessage;
```

Comment:
Why are we duplicating the message in storage?

Reply:
It's not stored -- the CreateAsync method is called before

Reply:
You are saving the message in the message repository as well as the session repository.

Reply:
Removed initialBotMessage from ChatSession object